Tiqr/Classes/AuthenticationConfirmViewController.m

```diff
@@ -43,15 +51,17 @@ @interface AuthenticationConfirmViewController ()
 @property (nonatomic, strong) IBOutlet UILabel *identityIdentifierLabel;
 @property (nonatomic, strong) IBOutlet UILabel *serviceProviderDisplayNameLabel;
 @property (nonatomic, strong) IBOutlet UILabel *serviceProviderIdentifierLabel;
+@property (nonatomic, copy) NSString *response;
+@property (strong, nonatomic) IBOutlet UIView *nonTouchIDViewsContainer;
 
 @end
 
 @implementation AuthenticationConfirmViewController
 
-- (instancetype)initWithAuthenticationChallenge:(AuthenticationChallenge *)challenge {
+- (instancetype)init {
     self = [super initWithNibName:@"AuthenticationConfirmView" bundle:nil];
 	if (self != nil) {
-		self.challenge = challenge;
+		self.challenge = ServiceContainer.sharedInstance.challengeService.currentAuthenticationChallenge;
```

Comment:
Why move from injection based (initWith) to state based (current... on the challengeService)? Isn't injection based more loosely coupled?